presto-hive/src/main/java/com/facebook/presto/hive/HiveWriterFactory.java

```diff
@@ -423,7 +433,8 @@ else if (insertExistingPartitionsBehavior == InsertExistingPartitionsBehavior.ER
 
         validateSchema(partitionName, schema);
 
-        String fileNameWithExtension = fileName + getFileExtension(conf, outputStorageFormat);
+        String extension = getFileExtension(conf, outputStorageFormat);
+        String fileNameWithExtension = stagingFileName + extension;
```

Comment:
I would simply remove the `fileNameWithExtension`, and would always add the extentsion to the `targetFileName` and `stagingFileName`